packages/page-coretime/src/utils/sale.ts

```diff
@@ -0,0 +1,244 @@
+// Copyright 2017-2024 @polkadot/app-coretime authors & contributors
```

Comment:
@filvecchiato you might wanna check out this file. It has a bunch of useful calcs :) Maybe we can collaborate together and combine our work and make a library of sorts with these functions.